packages/blocks/src/page-block/edgeless/components/toolbar/edgeless-toolbar.ts

```diff
@@ -159,9 +200,161 @@ export class EdgelessToolbar extends WithDisposable(LitElement) {
     }
   };
 
+  protected override updated(changedProperties: PropertyValues) {
+    const { type } = this.edgelessTool;
+    if (
+      changedProperties.has('_currentFrameIndex') &&
+      type === 'frameNavigator'
+    ) {
+      const frames = this.edgeless.surface.getElementsByType('frame');
```

Comment:
So when every time the toolbar gets updated, we need to iterate over all elements?

Reply:
uh, it's not good, I'll fix it.
since we can't add or delete frame on presentation, I'll cach frames